packages/rocketchat-api/server/v1/groups.js

```diff
@@ -348,7 +348,23 @@ RocketChat.API.v1.addRoute('groups.info', { authRequired: true }, {
 
 RocketChat.API.v1.addRoute('groups.invite', { authRequired: true }, {
 	post() {
-		const findResult = findPrivateGroupByIdOrName({ params: this.requestParams(), userId: this.userId });
+		let findResult;
+		const canAddUserToAnyPrivateGroup = RocketChat.authz.hasPermission(this.userId, 'add-user-to-any-p-room');
+		const params = this.requestParams();
+		if (canAddUserToAnyPrivateGroup) {
+			if (params.roomId && params.roomId.trim()) {
+				findResult = RocketChat.models.Subscriptions.findByRoomId(params.roomId).fetch()[0];
```

Comment:
Why not find the room directly? Looking for subscriptions will cause the code to fail on empty rooms.